client/v3/concurrency/session.go

```diff
@@ -45,9 +45,15 @@ func NewSession(client *v3.Client, opts ...SessionOption) (*Session, error) {
 		opt(ops, lg)
 	}
 
+	var cancel context.CancelFunc
+	sessionCreationCtx := ops.ctx
+	if ops.creationTimeout > 0 {
+		sessionCreationCtx, cancel = context.WithTimeout(ops.ctx, ops.creationTimeout)
+	}
```

Comment:
The `cancel` is overwritten by the following the other `cancel`
```suggestion
	var creationCancel context.CancelFunc
	sessionCreationCtx := ops.ctx
	if ops.creationTimeout > 0 {
		sessionCreationCtx, creationCancel = context.WithTimeout(ops.ctx, ops.creationTimeout)
		defer creationCancel()
	}
```